src/reconcile/rec_write.c

```diff
@@ -2743,6 +2739,8 @@ __rec_hs_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r)
         }
 
 err:
+    /* Reset the timer if we fail here before returning. */
+    session->reconcile_timeline.hs_wrapup_start = __wt_clock(session);
```

Comment:
I looked the code and I'm still confused why this is needed.

```
/*
     * If using the history store table eviction path and we found updates that weren't globally
     * visible when reconciling this page, copy them into the database's history store. This can
     * fail, so try before clearing the page's previous reconciliation state.
     */
    if (F_ISSET(r, WT_REC_HS)) {
        session->reconcile_timeline.hs_wrapup_start = __wt_clock(session);
        ret = __rec_hs_wrapup(session, r);
        session->reconcile_timeline.hs_wrapup_finish = __wt_clock(session);
        WT_RET(ret);
    }
```
We set the stop time no matter we hit an error or not. Why we would see a very large value?

Reply:
My theory is that even though we do stop the timer, when we hit an error, the error runs may be taking longer than we would expect a successful run to take - and that value from the error run still gets saved and inflates our numbers.

Reply:
I expect the opposite. The error runs should take much less time than the successful runs.

Reply:
After a chat with @quchenhao we have decided on a different fix, checking in similar areas that the finish time is bigger than the start time due to the possibility that machine time goes backwards due to time sync mechanisms. And we will take out this reset of the start time.